shading/coral-trino-parser/build.gradle

```diff
@@ -0,0 +1,26 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ConfigureShadowRelocation
+
+plugins {
+  id 'com.github.johnrengelman.shadow'
+}
+
+apply from: "$rootDir/gradle/java-publication.gradle"
+
+dependencies {
+  implementation 'io.trino:trino-parser:355'
```

Comment:
Could you convert this to `compileOnly`?

Reply:
Also the `shadowJar` task needs to package `compileOnly`:

```
shadowJar {
  configurations = [
    project.configurations.compileOnly
  ]
  dependsOn(relocateShadowJar)
}
```

Reply:
Any one change above will fail the local build of coral-trino:
```
> Task :coral-trino:compileJava FAILED
/Users/jiefli/code/coral-ljf-github/coral-trino/src/main/java/com/linkedin/coral/trino/trino2rel/PrestoParserDriver.java:8: error: package coral.shading.io.trino.sql.parser does not exist
import coral.shading.io.trino.sql.parser.ParsingOptions;
```

Reply:
You would need both.

Reply:
Thanks, it could build, do I need to open a PR with that change?

Reply:
Yes, please (preferably after testing through local integration with Trino).